src/include/hip/rpp_hip_common.hpp

```diff
@@ -72,4 +114,441 @@ inline RppStatus generate_gaussian_kernel_gpu(Rpp32f stdDev, Rpp32f* kernel, Rpp
+    src_f24->y.y = make_float4(rpp_hip_unpack1(src.y.y), rpp_hip_unpack0(src.z.x), rpp_hip_unpack3(src.z.x), rpp_hip_unpack2(src.z.y));
+
+    src_f24->z.x = make_float4(rpp_hip_unpack2(src.x.x), rpp_hip_unpack1(src.x.y), rpp_hip_unpack0(src.y.x), rpp_hip_unpack3(src.y.x));
+    src_f24->z.y = make_float4(rpp_hip_unpack2(src.y.y), rpp_hip_unpack1(src.z.x), rpp_hip_unpack0(src.z.y), rpp_hip_unpack3(src.z.y));
+}
+
+// F32 loads with layout toggle PKD3 to PLN3 (24 F32 pixels)
+
+__device__ __forceinline__ void rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(float *srcPtr, uint srcIdx, d_float24 *src_f24)
+{
+    src_f24->x.x.x = srcPtr[srcIdx];
+    src_f24->x.x.y = srcPtr[srcIdx + 3];
+    src_f24->x.x.z = srcPtr[srcIdx + 6];
+    src_f24->x.x.w = srcPtr[srcIdx + 9];
+    src_f24->x.y.x = srcPtr[srcIdx + 12];
+    src_f24->x.y.y = srcPtr[srcIdx + 15];
+    src_f24->x.y.z = srcPtr[srcIdx + 18];
+    src_f24->x.y.w = srcPtr[srcIdx + 21];
+
+    src_f24->y.x.x = srcPtr[srcIdx + 1];
+    src_f24->y.x.y = srcPtr[srcIdx + 4];
+    src_f24->y.x.z = srcPtr[srcIdx + 7];
+    src_f24->y.x.w = srcPtr[srcIdx + 10];
+    src_f24->y.y.x = srcPtr[srcIdx + 13];
+    src_f24->y.y.y = srcPtr[srcIdx + 16];
+    src_f24->y.y.z = srcPtr[srcIdx + 19];
+    src_f24->y.y.w = srcPtr[srcIdx + 22];
+
+    src_f24->z.x.x = srcPtr[srcIdx + 2];
+    src_f24->z.x.y = srcPtr[srcIdx + 5];
+    src_f24->z.x.z = srcPtr[srcIdx + 8];
+    src_f24->z.x.w = srcPtr[srcIdx + 11];
+    src_f24->z.y.x = srcPtr[srcIdx + 14];
+    src_f24->z.y.y = srcPtr[srcIdx + 17];
+    src_f24->z.y.z = srcPtr[srcIdx + 20];
+    src_f24->z.y.w = srcPtr[srcIdx + 23];
+}
+
+// I8 loads with layout toggle PKD3 to PLN3 (24 I8 pixels)
+
+__device__ __forceinline__ void rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(signed char *srcPtr, uint srcIdx, d_float24 *src_f24)
+{
+    d_uint6 src = *((d_uint6 *)(&srcPtr[srcIdx]));
+
+    src_f24->x.x = make_float4(rpp_hip_unpack0(src.x.x), rpp_hip_unpack3(src.x.x), rpp_hip_unpack2(src.x.y), rpp_hip_unpack1(src.y.x));
+    src_f24->x.y = make_float4(rpp_hip_unpack0(src.y.y), rpp_hip_unpack3(src.y.y), rpp_hip_unpack2(src.z.x), rpp_hip_unpack1(src.z.y));
+
+    src_f24->y.x = make_float4(rpp_hip_unpack1(src.x.x), rpp_hip_unpack0(src.x.y), rpp_hip_unpack3(src.x.y), rpp_hip_unpack2(src.y.x));
+    src_f24->y.y = make_float4(rpp_hip_unpack1(src.y.y), rpp_hip_unpack0(src.z.x), rpp_hip_unpack3(src.z.x), rpp_hip_unpack2(src.z.y));
+
+    src_f24->z.x = make_float4(rpp_hip_unpack2(src.x.x), rpp_hip_unpack1(src.x.y), rpp_hip_unpack0(src.y.x), rpp_hip_unpack3(src.y.x));
+    src_f24->z.y = make_float4(rpp_hip_unpack2(src.y.y), rpp_hip_unpack1(src.z.x), rpp_hip_unpack0(src.z.y), rpp_hip_unpack3(src.z.y));
+}
+
+// F16 loads with layout toggle PKD3 to PLN3 (24 F16 pixels)
+
+__device__ __forceinline__ void rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(half *srcPtr, uint srcIdx, d_float24 *src_f24)
+{
+    src_f24->x.x.x = __half2float(srcPtr[srcIdx]);
```

Comment:
use the same comments mentioned above for rpp_hip_load24_pkd3_and_unpack_to_float24_pln3 function to rewrite this function as well.

Reply:
Done